sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -573,6 +637,71 @@ case class CatalogTable(
     map
   }
 
+  def toJsonLinkedHashMap: mutable.LinkedHashMap[String, String] = {
+    val map = new mutable.LinkedHashMap[String, String]()
+
+    val filteredTableProperties = SQLConf.get
+      .redactOptions(properties.filter { case (k, v) =>
+        !k.startsWith(VIEW_PREFIX) && v.nonEmpty
+      })
+
+    val tableProperties: String =
+      if (filteredTableProperties.isEmpty) ""
+      else filteredTableProperties.toSeq
+        .sortBy(_._1)
+        .map { case (k, v) => s""""$k": "$v"""" }
+        .mkString("{", ", ", "}")
+
+    val partitionColumns = partitionColumnNames
+      .map(s => s""""$s"""")
+      .mkString("[", ", ", "]")
+
+    val lastAccess =
+      if (lastAccessTime <= 0) "\"UNKNOWN\""
+      else s""""${new Date(lastAccessTime).toInstant.toString}""""
```

Comment:
Could you avoid using Java7 datetime classes, please and use Java 8 instances directly.